kobo/apps/accounts/adapter.py

```diff
@@ -19,3 +20,11 @@ def save_user(self, request, user, form, commit=True):
             if commit:
                 user.extra_details.save()
         return user
+
+
+class SocialAccountAdapter(DefaultSocialAccountAdapter):
+    def populate_user(self, request, sociallogin, data):
+        user = super().populate_user(request, sociallogin, data)
+        if sociallogin.email_addresses:
```

Comment:
I think this code deserves a comment about how it's an upstream bug. Can you open an upstream bug report and link in the comment. That will make it easier to remove/refactor later.

This only happens with OIDC, correct? Could we check if user.email exists here as well. There is no need to set the email again. It probably doesn't matter unless there is some flow where it would be different or the new populated user has multiple emails (impossible at this time, I think).

Reply:
@bufke It opened an issue, but it looks like there is another issue that already addresses this: https://github.com/pennersr/django-allauth/issues/3241. It already has an associated PR that will fix this issue. I'll update the comment in the code here to point to that issue